keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,854 @@
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We support the `managed-by` label as a lightweight mechanism to delegate the Job
+synchronization to an external controller.
+
+## Motivation
+
+As a part of [Kueue](https://github.com/kubernetes-sigs) (an effort done by
+Batch-WG, in cooperation with SIG-Autoscaling, SIG-Scheduling, SIG-Apps and SIG-Node) we
+are working on a multi-cluster job dispatcher project, called
+[MultiKueue](https://github.com/kubernetes-sigs/kueue/pull/1380).
+
+In the MultiKueue design, which follows manager-worker architecture, a user
+creates a Job in the management cluster, but a mirror-copy of the Job is created
+and executed in one of the worker clusters. The status updates of the mirror-Job
+are reflected by the Kueue controller in the management cluster, in the status
+of the Job created by the user.
+
+In order to support this workflow we need a mechanism to disable the main Job
+controller, and delegate the status synchronization to the Kueue controller.
+
+### Goals
+
+- support delegation of Job synchronization to an external controller
+
+### Non-Goals
+
+- passing custom parameters to the external controller
+
+## Proposal
+
+The proposal is to support the `managed-by` label to disabled the Job controller
+from synchronizing jobs with the annotation.
+
```

Comment:
Some questions:

- Can this label match more than one controller? E.g., if you have two controllers that are separate with the same name, both get the job mirrored there? And then the first one to schedule gets the final assignment?
- What happens with a JobSet if the Jobs inside get different `managed-by` labels? As long as they don't share a common network or closely work together, that might be a valid use case, but I haven't thought about it more than that.

Reply:
> Can this label match more than one controller? E.g., if you have two controllers that are separate with the same name, both get the job mirrored there? And then the first one to schedule gets the final assignment?

In principle there is only one Job controller, indicated by the label value. I don't think there are any means in the core k8s to verify this is the case though.

> What happens with a JobSet if the Jobs inside get different managed-by labels? As long as they don't share a common network or closely work together, that might be a valid use case, but I haven't thought about it more than that.

Yes, if you have a JobSet with two sub-jobs, the sub-jobs could use different controllers, technically, yes. I think it is up to users / framework developers to figure out if this is useful for their use-cases.